packages/cozy-stack-client/src/PermissionCollection.spec.js

```diff
@@ -180,4 +259,27 @@ describe('getPermissionsFor', () => {
     const perms2 = getPermissionsFor(document, true, options)
     expect(perms2.files.verbs).toEqual(expect.arrayContaining(verbs))
   })
+
+  it('Should allow other doctypes', () => {
+    const document = { _type: 'io.cozy.notes', _id: '1234' }
+    const perms = getPermissionsFor(document, false)
+    expect(perms.files.type).toEqual('io.cozy.notes')
+  })
+})
+
+describe('isCollection', () => {
+  it('io.cozy.file is not a collection', () => {
+    const document = { _type: 'io.cozy.files', _id: '1234' }
+    expect(isCollection(document)).toBe(false)
+  })
+
+  it('io.cozy.notes.events is not a collection', () => {
+    const document = { _type: 'io.cozy.notes.events', _id: '1234' }
+    expect(isCollection(document)).toBe(false)
+  })
+
+  it('io.cozy.photos.albums is a collection', () => {
+    const document = { _type: 'io.cozy.photos.albums', _id: '1234' }
+    expect(isCollection(document)).toBe(true)
+  })
```

Comment:
I am wondering if isCollection should be tested explicitly ? AMHA it should be tested as part of the `createSharingLink` tests. Testing internal methods makes test brittle as we cannot easily rename methods (breaking the tests).

Reply:
I usually like to test methods that have a real functional meaning. If we delete it later, we can easily delete the associated tests, but this garantee the functional.

This is less true for purely technical helpers.

isCollection (renamed now) is in the former case